tensorflow/core/kernels/data/captured_function.cc

```diff
@@ -897,7 +906,24 @@ Status InstantiatedCapturedFunction::RunWithBorrowedArgs(
             f_opts.step_id, "#");
       },
       profiler::TraceMeLevel::kInfo);
-  TF_RETURN_IF_ERROR(lib_->RunSync(std::move(f_opts), f_handle_, &frame));
+  if (node) {
```

Comment:
Add a `TODO(jsimsa)` for factoring out the common code between `RunAsync`, `Run`, and `RunWithBorrowedArguments` to a function.